projects/overlook.md

```diff
@@ -143,76 +166,118 @@ password: overlook2021
-* 2: Class methods use a mix of array and object prototypes and for loops. Application runs but the code has long methods, unnecessary or poorly named variables, and needs significant refactoring.  Uses `fetch` effectively for `GET` but does not implement `POST`.  Has little error handling and only `console logs` errors if a network request fails.
-* 1:  Application generates syntax error or crashes during execution.  Application is not separated into classes and there is no separation of business-side logic and view-related code. Developer writes code with unnecessary variables, operations, or steps that do not increase clarity.
+✨WOW✨ can look like: 
 
+* Design is responsive across small, medium and large breakpoints
+* Special consideration was made around accessibility
+</section>
 
-## Testing
+<section class="answer">
+### Fundamental JavaScript and Style / Fetch Competency Examples
+
+* Code is divided into logical components each with a clean, single responsibility
+* Array prototype methods are used to iterate instead of for loops
+* All DOM manipulation is held in the scripts.js file. No DOM manipulation occurs outside of this file.
+* Variables and functions are consistently and appropriately named
+* Code leverages JavaScript's truthy/falsey principles
+* Demonstrates efforts towards making functions pure when possible. **Note: Purity is not possible for every function in a FE application. Strive for it only when it makes sense.**
+* Implements GET and POST Fetch requests
+* When utilizing Fetch, the DOM is updated based on the results of that Fetch
+* Most errors are handled and messages are displayed to a user when an error occurs
+
+✨WOW✨ can look like: 
+
+* All data manipulation is cleanly separated from rendering code. 
+* Code has clearly been refactored; functions are DRY and adhere to the Single Responsibility Principle. 
+* All errors are handled and messages are displayed to a user when an error occurs
+* Effectively implements one or more closure throughout project.
+</section>
+
+<section class="answer">
+### Testing
+
+* Tests cover most features and test for happy paths
+* Test suite is organized.
+* Each function is tested in its own it block.
+* Each path is tested in its own it block.
+* Rather than using the production data, small sample data is stored in its own file and used for testing.
+* Sample data has been crafted to create the scenarios needed for thorough testing.
+* beforeEach hook is used to DRY up test files
+* There are no failing/pending tests upon submission 
+
+✨WOW✨ can look like: 
+
+* All scenarios/outcomes/paths are tested for functions, including happy and sad paths.
+* Application has a robust and thorough test suite that covers all functions that do not update the dom.
```

Comment:
I'd like to see sad path testing included in the standard competencies rather than listed as a WOW.

Something I struggled with when updating the WC rubric was that separating out the WOWs like this feels a little like lowering the bar to me.  I really want them to strive to test ALL scenarios and paths and to have a robust and thorough test suite that covers all function.   I want that to be the bar they're going for, not something they think of as an above-and-beyond WOW.   Even if they're striving for that bar, it is likely some things are missing and they'll get a YES instead of a WOW.   When the WOWs are split out like this I worry they can think of sad paths as optional (for example).

For Mod 2, my preference would be that we list competency examples that would make for great code, then, based on how well/thoroughly they're able to accomplish those things, they can be scored as Yes or Wow.  
If they do a pretty good job, they meets our expectations and it's a YES.  
If they 100% crush it, they exceed our expectations and it's a WOW.    My thinking is that for Mod 2 rubrics we would only note specific WOW bullets if it's a specific extra thing they did, like responsive design.   Otherwise we would determine Yes vs Wow with the degree of competency/excellence they show on any given competency bullet.

This does veer (evolve?) from the explicit breakdown of the Mod 1 rubrics but I think opens up more opportunity for them to strive for excellence without any added pressure - as well as simplifying/condensing the rubrics a bit.   I'd love to hear some other instructors weigh in on this.  We probably want consistency across the Mod 2 projects on this.

Reply:
I think we should either open up conversation around this, since updating this would mean adding more requirements for students for these projects. This section is based on 2s and 3s vs 4s in the original rubric and students have not had to include sad paths to get a 3 on these projects before. Personally, I don't think we should add anything without taking something else out - but I have no idea what we would want to take out. 

The other option is to leave the expectations as they were, let the M2 team feel out the new functional expectations along with the new rubric expectations and _then_ decide if they'd like to add in more requirements. 

Let me know if you'd like this to be brought to discussion!

Reply:
I'd love to bring this to discussion.  Its definitely a place where we want the team to be aligned on what the expectations are for students and for grading.  

I would really love to see us NOT leave the expectations as they were.  For a long time I've felt the M2 rubrics needed an overhaul which is partly why I started drafting changes to them but that got paused when I left M2 and we started talking about FP.  

Since we're outlining the competency examples as examples of what to strive for, rather than a checklist of requirements, I don't really even think of it as adding requirements, other than expecting sad paths which would be an added expectation compared to before.  
I think of it more as saying "here is what a great project would look like, do you best!"   
Whereas the way we've broken it out before feels more like we're saying "you really only have to do a partial job to pass" which makes me feel like students could not even strive for the best they could do.

Reply:
Sweet! Thanks for getting that conversation going. I'm glad you brought it up. 

Based on the conversation, I made some changes. I also made some changes according to Robbie's checklist thoughts because they really resonated with me - but I'm happy to change those back, if we need.

Reply:
I think this all looks great!  Thanks for the discussion and brainstorming on the call and thanks for this beautiful work!